runelite-client/src/test/java/net/runelite/client/plugins/xpglobes/XpGlobesTest.java

```diff
@@ -0,0 +1,181 @@
+	}
+
+	@Test
+	public void testVirtualLevelInGlobeIsShownWhenConfigured()
+	{
+		int totalXp = 15_000_000;
+		when(xpGlobesConfig.showVirtualLevel()).thenReturn(true);
+		runFirstStatChangeForCaching(totalXp);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			totalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Experience.getLevelForXp(totalXp + 1), xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testGlobeIsNotShownWhenHideMaxAndShowVirtualLevelConfigured()
+	{
+		int totalXp = 15_000_000;
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+		runFirstStatChangeForCaching(totalXp);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			totalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
+	}
+
+	@Test
+	public void testGlobeIsNotShownWhenHideMaxConfigured()
+	{
+		int totalXp = 15_000_000;
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+		runFirstStatChangeForCaching(totalXp);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			totalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
+	}
+
+	@Test
+	public void testGlobeIsShownOnXpGainBelowMaxWhenHideMaxConfigured()
+	{
+		int totalXp = 1;
+		when(xpGlobesConfig.showVirtualLevel()).thenReturn(true);
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+		runFirstStatChangeForCaching(totalXp);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			totalXp + 150,
+			Experience.getLevelForXp(totalXp + 150),
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Experience.getLevelForXp(totalXp + 150), xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testStatChangesFromBoostDoNotAffectXpGlobes()
+	{
+		int total_xp = 9_000_000;
+		runFirstStatChangeForCaching(total_xp);
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
+
+		StatChanged boostEvent = new StatChanged(
+			Skill.AGILITY,
+			total_xp,
+			Experience.getLevelForXp(total_xp),
+			5
+		);
+
+		// XP hasn't changed since the first event, so no globe should be added
+		xpGlobesPlugin.onStatChanged(boostEvent);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+	}
+
+	void runFirstStatChangeForCaching(int totalXp)
```

Comment:
Invoke this in `before()` instead of calling it in each test case

Reply:
done